packages/table/src/table.tsx

```diff
@@ -1191,8 +1197,10 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         return this.state.verticalGuides != null || this.state.horizontalGuides != null;
     }
 
-    private isSelectionModeEnabled(selectionMode: RegionCardinality) {
-        return this.props.selectionModes.indexOf(selectionMode) >= 0;
+    private isSelectionModeEnabled(selectionMode: RegionCardinality, selectionModes = this.props.selectionModes) {
+        const { numRows } = this.props;
+        const numColumns = this.childrenArray.length;
+        return selectionModes.indexOf(selectionMode) >= 0 && numRows > 0 && numColumns > 0;
```

Comment:
A selection mode is now "enabled" iff is not disabled _and_ if the table has body cells.